bincrafters/build_shared.py

```diff
@@ -161,8 +161,11 @@ def get_conan_remotes(username):
     return remotes
 
 
-def get_upload_when_stable():
-    return get_bool_from_env("CONAN_UPLOAD_ONLY_WHEN_STABLE")
+def get_upload_when_stable(kwargs):
+    upload_when_stable = kwargs.get('upload_only_when_stable')
+    if upload_when_stable is None:
+        kwargs['upload_only_when_stable'] = get_bool_from_env("CONAN_UPLOAD_ONLY_WHEN_STABLE")
```

Comment:
does it apply to only `upload_only_when_stable` parameter? I suspect others might be affected by the same bug?

Reply:
you are right, but we would need to update dozen of arguments. As bincrafters_package_tools is not common for other users, I think only updating what we really need is enough, otherwise we would add complexity to our code and it would not be used.

Reply:
wait, thinking again, we have few parameters, I think worth the effort. I'll support other arguments too.

Reply:
Done